model/boundary.F90

```diff
@@ -2248,6 +2248,8 @@ subroutine update_coarse_grid_mpp_2d(var_coarse, var_nest, nest_domain, dx, dy,
 
    real :: var_nest_3d(is_n:ie_n+istag,js_n:je_n+jstag,1)
    real :: var_coarse_3d(isd_p:ied_p+istag,jsd_p:jed_p+jstag,1)
+   integer( KIND = 8) :: ptr_nest=0
+   integer( KIND = 8) :: ptr_coarse=0
```

Comment:
Do these have to be hard coded to be `kind=8`?

Reply:
Because it is being used to actually point to a memory location provided via the LOC function, it needs to be a 64-bit integer.